diagnostic.js

```diff
@@ -32,25 +32,72 @@ const Movement = require('./models/movement.js');
 
 /// ADD YOUR CODE BELOW
 
-const create = (name, description, startYear, endYear) => {};
+const create = (name, description, startYear, endYear) => {
+  Movement.create({
+    name: name,
+    description: description,
+    startYear: startYear,
+    endYear: endYear
+  }).then(function(movement) {
+    console.log(movement.toJSON());
+  }).then(console.error);
+};
 // Success -> console.log new Movement as JSON
 // Failure -> Console.error
 
-const index = () => {};
+const index = () => {
+let search = {};
+if (arguments[0] && arguments[1]) {
+    let field = arguments[0];
+    let criterion = arguments[1];
+    if (criterion[0] === '/') {
+      let regex = new RegExp(criterion.slice(1, criterion.length -1));
+      search[field] = regex;
+    } else {
+      search[field] = criterion;
+    }
+  }
+
+  Movement.find(search).then(function(movement) {
+    movement.forEach(function(movement) {
+      console.log(movement.toJSON());
+    });
+  }).catch(console.error);
+};
 // Success -> console.log all Movements as JSON
 // Failure -> Console.error
 
-const show = (id) => {};
+const show = (id) => {
+  Movement.findById(id).then(function(movement) {
+    console.log(movement.toObject());
+  }).catch(console.error).catch(console.log('Not Found'));
+};
 // Success -> If the specified Movement exists, console.log it as JSON;
 //              otherwise, console.log "Not Found" and exit.
 // Failure -> Console.error
 
-const update = (id, field, value) => {};
+const update = (id, field, value) => {
+  let modify = {};
+    modify[field] = value;
+    Movement.findById(id).then(function(movement) {
+      movement.set(field, value);
+      return movement.save();
+    }).then(function(movement) {
+      console.log(movement.toJSON());
+    }).catch(console.error)
+    .catch(console.log('Not Found'));
+};
 // Success -> If the specified Movement exists, update it and console.log the
 //              updated Movement as JSON; otherwise, console.log "Not Found" and exit.
 // Failure -> Console.error
 
-const destroy = (id) => {};
+const destroy = (id) => {
+  Movement.findById(id).then(function(movement) {
+    return movement.remove();
```

Comment:
The existence of the document should be checked before attempting to use it, otherwise an error will be thrown.  You also don't need two calls to `catch()`.

```javascript
const destroy = id => {
  console.log('destroy');
  Movement.deleteOne({ id })
    .then(movement => {
      if (!movement) {
        console.log('Not found.');

        return null;
      }

      movement.remove();

      console.log('Removed.');

      return null;
    })
    .catch(console.error);
};
```